packages/dev/core/src/Culling/Helper/computeShaderBoundingHelper.ts

```diff
@@ -0,0 +1,168 @@
+            // const boneSampler = mesh.skeleton.getTransformMatrixTexture(mesh);
+            // computeShader.setTexture("boneSampler", boneSampler!, false);
+            // if (mesh.numBoneInfluencers > 4) {
+            //     this._extractDataAndLink(computeShader, mesh as Mesh, VertexBuffer.MatricesIndicesExtraKind, 4, "indexExtraBuffer", this._indexExtraBuffers);
+            //     this._extractDataAndLink(computeShader, mesh as Mesh, VertexBuffer.MatricesWeightsExtraKind, 4, "weightExtraBuffer", this._weightExtraBuffers);
+            // }
+        }
+
+        // Results
+        let resultData: Float32Array;
+        if (!this._resultData[mesh.uniqueId]) {
+            resultData = new Float32Array(6);
+
+            this._resultData[mesh.uniqueId] = resultData;
+        } else {
+            resultData = this._resultData[mesh.uniqueId];
+        }
+
+        resultData[0] = Number.POSITIVE_INFINITY;
+        resultData[1] = Number.POSITIVE_INFINITY;
+        resultData[2] = Number.POSITIVE_INFINITY;
+
+        resultData[3] = Number.NEGATIVE_INFINITY;
+        resultData[4] = Number.NEGATIVE_INFINITY;
+        resultData[5] = Number.NEGATIVE_INFINITY;
+
+        let resultBuffer: StorageBuffer;
+        if (!this._resultBuffers[mesh.uniqueId]) {
+            resultBuffer = new StorageBuffer(this._engine as WebGPUEngine, Float32Array.BYTES_PER_ELEMENT * 6);
+
+            this._resultBuffers[mesh.uniqueId] = resultBuffer;
+        } else {
+            resultBuffer = this._resultBuffers[mesh.uniqueId];
+        }
+
+        resultBuffer.update(resultData);
+
+        computeShader.setStorageBuffer("resultBuffer", resultBuffer);
+
+        // Dispatch
+        return new Promise((resolve) => {
+            computeShader.dispatchWhenReady(Math.ceil(vertexCount / 64)).then(() => {
+                resultBuffer.read(undefined, undefined, resultData, true).then(() => {
+                    mesh._refreshBoundingInfoDirect({
+                        minimum: Vector3.FromArray(resultData, 0),
+                        maximum: Vector3.FromArray(resultData, 3),
+                    });
+                    resolve();
+                });
+            });
+        });
+    }
+
+    private _disposeCache(storageUnit: { [key: number]: StorageBuffer }) {
+        for (const key in storageUnit) {
+            storageUnit[key].dispose();
+        }
+    }
+
+    /** @internal */
+    public dispose(): void {
+        this._disposeCache(this._positionBuffers);
+        this._positionBuffers = {};
+        this._disposeCache(this._indexBuffers);
+        this._indexBuffers = {};
+        this._disposeCache(this._weightBuffers);
+        this._weightBuffers = {};
+        this._disposeCache(this._resultBuffers);
+        this._resultBuffers = {};
+        this._resultData = {};
+        this._computeShaders = {};
+        this._engine = null;
+    }
+}
+
+RegisterClass("BABYLON.ComputeShaderBoundingHelper", ComputeShaderBoundingHelper);

```

Comment:
It looks like this is introducing new dependencies on side effects. Do we really a want to do that?

Reply:
Yes but we do not have another option here. Else the BoundingInfoHelper will need to import both platforms

Reply:
The dynamic import you changed it to seems like a great solution if we are ready to start using dynamic imports!

I think other alternatives could have been:
- Have an `IBoundingInfoHelper` with three implementations: `TransformFeedbackBoundingInfoHelper`, `ComputeShaderBoundingInfoHelper`, and `UniversalBoundingInfoHelper` (imports the other two). Then the user can choose whether it is more important to them to have it "just work" or whether they prefer to only pull in the one they know they need. This option doesn't really work if it is used in core code or library code (not final user code).
- Expose the functionality off of `Engine` (so WebGL, WebGPU, and Native can each provide a solution). This would probably be ok in core/library code since we already have the `Engine` abstraction.

Reply:
Yes I checked with raanan and Seb and it should be all good